zipline/utils/cache.py

```diff
@@ -298,9 +304,25 @@ def __enter__(self):
         return self
 
     def __exit__(self, *exc_info):
+        self._tmpfile.__exit__(*exc_info)
         if exc_info[0] is None:
             self._commit()
-        self._tmpfile.__exit__(*exc_info)
+
+    @property
```

Comment:
No reason in particular; if the tmpfile isn't necessarily a closeable file, then I wanted to be able to handle that in the future. 

In this case, the file is actually an SQLite3 Connection object. I figured to add these just in case there was some unforeseeable object type that the working file needed to use.